crates/core/database/src/models/channels/ops/mongodb.rs

```diff
@@ -0,0 +1,190 @@
+use super::AbstractChannels;
+use crate::{Channel, FieldsChannel, IntoDocumentPath, MongoDb, PartialChannel};
+use bson::Document;
+use futures::{FutureExt, StreamExt};
+use revolt_permissions::OverrideField;
+use revolt_result::Result;
+static COL: &str = "channels";
+
+#[async_trait]
+impl AbstractChannels for MongoDb {
+    /// Insert a new channel in the database
+    async fn insert_channel(&self, channel: &Channel) -> Result<()> {
+        query!(self, insert_one, COL, &channel).map(|_| ())
+    }
+
+    /// Insert a a user to a group
+    async fn add_user_to_group(&self, channel: &str, user: &str) -> Result<()> {
+        let user_doc = doc! {
+            "_id": channel
+        };
+        let group_doc = doc! {
+            "$push": {
+                "recipients": user
+            }
+        };
+        Ok(self
+            .col::<Document>(COL)
+            .update_one(user_doc, group_doc, None)
```

Comment:
You can put the `doc! {}` macro straight into the method parameters, this is how I've been doing it so far on the other collections.